server_tgi.py

```diff
@@ -0,0 +1,284 @@
+import argparse
+import json
+import logging
+import time
+import uuid
+from contextlib import asynccontextmanager
+from pathlib import Path
+from typing import AsyncGenerator, Optional, Tuple
+
+import docker
+import requests
+import uvicorn
+from fastapi import FastAPI, Request
+from fastapi.responses import StreamingResponse
+from huggingface_hub import InferenceClient
+from transformers import AutoTokenizer
+
+from functionary.inference_stream import generate_openai_format_from_stream_async
+from functionary.openai_types import (
+    ChatCompletionChunk,
+    ChatCompletionRequest,
+    ChatCompletionResponse,
+    ChatCompletionResponseChoice,
+    ChatMessage,
+    Function,
+    StreamChoice,
+    Tool,
+    UsageInfo,
+)
+from functionary.prompt_template import get_prompt_template_from_tokenizer
+from functionary.prompt_template.prompt_utils import enforce_tool_choice
+
+logging.basicConfig(level=logging.INFO)
```

Comment:
Can you put this into if __name__ clause so it doesn't modify the logger if imported from another module